src/common/kube-helpers.ts

```diff
@@ -0,0 +1,162 @@
+ * - User
+ * - Cluster
+ * - Context
+ * @param config KubeConfig to check
+ */
+export function validateConfig(config: KubeConfig | string): KubeConfig {
+  if (typeof config == "string") {
+    config = loadConfig(config);
+  }
+  logger.debug(`validating kube config: ${JSON.stringify(config)}`)
+  if (!config.users || config.users.length == 0) {
+    throw new Error("No users provided in config")
+  }
+  if (!config.clusters || config.clusters.length == 0) {
+    throw new Error("No clusters provided in config")
+  }
+  if (!config.contexts || config.contexts.length == 0) {
+    throw new Error("No contexts provided in config")
+  }
+
+  return config
+}
+
+/**
+ * Breaks kube config into several configs. Each context as it own KubeConfig object
+ */
+export function splitConfig(kubeConfig: KubeConfig): KubeConfig[] {
+  const configs: KubeConfig[] = []
+  if (!kubeConfig.contexts) {
+    return configs;
+  }
+  kubeConfig.contexts.forEach(ctx => {
+    const kc = new KubeConfig();
+    kc.clusters = [kubeConfig.getCluster(ctx.cluster)].filter(n => n);
+    kc.users = [kubeConfig.getUser(ctx.user)].filter(n => n)
+    kc.contexts = [kubeConfig.getContextObject(ctx.name)].filter(n => n)
+    kc.setCurrentContext(ctx.name);
+
+    configs.push(kc);
+  });
+  return configs;
+}
+
+export function dumpConfigYaml(kubeConfig: Partial<KubeConfig>): string {
+  const config = {
+    apiVersion: "v1",
+    kind: "Config",
+    preferences: {},
+    'current-context': kubeConfig.currentContext,
+    clusters: kubeConfig.clusters.map(cluster => {
+      return {
+        name: cluster.name,
+        cluster: {
+          'certificate-authority-data': cluster.caData,
+          'certificate-authority': cluster.caFile,
+          server: cluster.server,
+          'insecure-skip-tls-verify': cluster.skipTLSVerify
+        }
+      }
+    }),
+    contexts: kubeConfig.contexts.map(context => {
+      return {
+        name: context.name,
+        context: {
+          cluster: context.cluster,
+          user: context.user,
+          namespace: context.namespace
+        }
+      }
+    }),
+    users: kubeConfig.users.map(user => {
+      return {
+        name: user.name,
+        user: {
+          'client-certificate-data': user.certData,
+          'client-certificate': user.certFile,
+          'client-key-data': user.keyData,
+          'client-key': user.keyFile,
+          'auth-provider': user.authProvider,
+          exec: user.exec,
+          token: user.token,
+          username: user.username,
+          password: user.password
+        }
+      }
+    })
+  }
+
+  logger.debug("Dumping KubeConfig:", config);
+
+  // skipInvalid: true makes dump ignore undefined values
+  return yaml.safeDump(config, { skipInvalid: true });
+}
+
+export function podHasIssues(pod: V1Pod) {
+  // Logic adapted from dashboard
+  const notReady = !!pod.status.conditions.find(condition => {
+    return condition.type == "Ready" && condition.status !== "True"
```

Comment:
`condition.type === "Ready"`?

Reply:
we don't have strict usage of this (btw code moved here from another place, I prefer ===)